src/hotspot/share/memory/metaspace/binList.hpp

```diff
@@ -181,12 +179,12 @@ class BinListImpl {
   void verify() const {
     MemRangeCounter local_counter;
     for (int i = 0; i < num_lists; i++) {
-      const size_t s = MinWordSize + i;
+      const size_t s = word_size_for_index(i);
       int pos = 0;
       for (Block* b = _blocks[i]; b != nullptr; b = b->_next, pos++) {
-        assert(b->_word_size == s,
-               "bad block size in list[%d] at pos %d (" BLOCK_FORMAT ")",
-               i, pos, BLOCK_FORMAT_ARGS(b));
+        assert(s == 1 || ((MetaWord*)b)[s - 1] == 0,
```

Comment:
Again isn't the first case handled by the second?

Reply:
See answer above. I realize I probably should factor this out into a reasonably named method.